corehq/apps/case_importer/tracking/filestorage.py

```diff
@@ -17,21 +18,27 @@ def __init__(self, bucket, meta_model):
             must contain columns identifier, filename, length
         """
 
-        self._bucket = bucket
+        self._bucket_prefix = bucket
         self._db = get_blob_db()
         self._meta_model = meta_model
 
+    def _get_bucket(self, padding):
+        return '{}/{}'.format(self._bucket_prefix, padding)
+
     def write_file(self, f, filename):
-        blob_info = self._db.put(f, bucket=self._bucket)
-        self._meta_model(identifier=blob_info.identifier, filename=filename,
+        padding = random_url_id(16)
+        blob_info = self._db.put(f, bucket=self._get_bucket(padding))
+        identifier = '{}/{}'.format(padding, blob_info.identifier)
+        self._meta_model(identifier=identifier, filename=filename,
```

Comment:
@millerdev and I just had a good talk about blobdb interface and how I can very simply improve upon it to not have to have these weird '{my-128-bit}/{blob-db-short-id}' keys.

The outcome is that I'll change blobdb.put to accept an `identifier`, and if it's not passed in, to default to the current behavior of auto-generating short ids. I'll also use the currently unused `basename` param, being careful to make sure it's not passed by any callers. Then @millerdev will have it on his plate to (at some time in the future) make `identifier` a required field so that the default isn't using short ids that result in a collision if your bucket is just a static namespace, and changing everywhere that uses that functionality to generate their own short ids.